pdr_backend/pdr_dashboard/test/test_callbacks_feeds.py

```diff
@@ -0,0 +1,231 @@
+import json
+import time
+
+from selenium.webdriver.common.by import By
+from pdr_backend.pdr_dashboard.test.resources import (
+    _navigate_to_feeds_page,
+    _assert_table_row_count,
+    start_server_and_wait,
+    _clear_filters,
+    _set_dropdown_and_verify_row_count,
+    _set_input_value_and_submit,
+    _remove_tags,
+    _set_searchbar_value,
+)
+
+
+def _get_table_data_as_json(table):
+    rows = table.find_elements(By.XPATH, ".//tr")
+    header = rows[0].find_elements(By.XPATH, ".//th")
+
+    headers = [_remove_tags(th.text) for th in header]
+
+    table_data = []
+
+    for row in rows[1:]:  # Skip header row
+        cells = row.find_elements(By.XPATH, ".//td")
+        row_data = {headers[i]: _remove_tags(cells[i].text) for i in range(len(cells))}
+        table_data.append(row_data)
+
+    return table_data
+
+
+def _save_table_data_as_json(table, filename):
+    table_data = _get_table_data_as_json(table)
+    with open("pdr_backend/pdr_dashboard/test/" + filename, "w") as f:
+        json.dump(table_data, f, indent=2)
+
+
+def _verify_table_data(table, filename):
+    table_data = _get_table_data_as_json(table)
+    with open("pdr_backend/pdr_dashboard/test/" + filename) as f:
+        expected_data = json.load(f)
+
+    for _, row in enumerate(table_data):
+        assert row in expected_data
+
+
+def test_feeds_table(_sample_app, dash_duo):
+    app = _sample_app
+    start_server_and_wait(dash_duo, app)
+
+    _navigate_to_feeds_page(dash_duo)
+    dash_duo.wait_for_element("#feeds_page_table table")
+
+    table = dash_duo.find_element("#feeds_page_table table")
+
+    # Validate row and column count
+    rows = table.find_elements(By.XPATH, ".//tr")
+    assert len(rows) == 21
+
+    header = rows[0].find_elements(By.XPATH, ".//th")
+    columns = header
+    assert len(columns) == 12
+
+    # Validate headers
+    header_texts = [_remove_tags(c.text) for c in columns]
+    expected_headers = [
+        "",
```

Comment:
this header line is a bit of a pink flag to me. Could you help me understand what is happening here?

Reply:
The first column contains the radio inputs for selecting rows, that's why the header is empty